stock_account_valuation_report/models/product_product.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2020 ForgeFlow S.L.
+# Copyright 2019 Aleph Objects, Inc.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import _, api, fields, models
+from odoo.tools import float_compare
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    stock_value = fields.Float("Inventory Value", compute="_compute_inventory_value")
+    account_value = fields.Float("Accounting Value", compute="_compute_inventory_value")
+    qty_at_date = fields.Float("Inventory Quantity", compute="_compute_inventory_value")
+    account_qty_at_date = fields.Float(
+        "Accounting Quantity", compute="_compute_inventory_value"
+    )
+    stock_fifo_real_time_aml_ids = fields.Many2many(
+        "account.move.line", compute="_compute_inventory_value"
+    )
+    stock_valuation_layer_ids = fields.Many2many(
+        "stock.valuation.layer", compute="_compute_inventory_value"
+    )
+    valuation_discrepancy = fields.Float(
+        compute="_compute_inventory_value",
+        search="_search_valuation_discrepancy",
+    )
+    qty_discrepancy = fields.Float(
+        compute="_compute_inventory_value",
+        search="_search_qty_discrepancy",
+    )
+
+    @api.model
+    def _search_qty_discrepancy(self, operator, value):
+        products = self.with_context(active_test=False).search(
+            [("is_storable", "=", True)],
+        )
+        dp = self.env["decimal.precision"].precision_get("Product Price")
```

Comment:
This looks wrong, price != quantity, I think it should be UoM precision, I made similar changes in my recent PR for discrepancy adjustment, use UoM digits here

Reply:
`pp.uom_id.rounding`?

Reply:
See new commit